lib/bundler/settings.rb

```diff
@@ -289,11 +289,7 @@ def load_config(config_file)
     def self.normalize_uri(uri)
       uri = uri.to_s
       uri = "#{uri}/" unless uri =~ %r{/\Z}
-      uri = URI(uri)
-      unless uri.absolute?
-        raise ArgumentError, "Gem sources must be absolute. You provided '#{uri}'."
-      end
-      uri
+      URI(uri)
```

Comment:
I think this will completely change the way we validate any existing source URLs, and that's probably not okay. Can we do the host extraction in a different method so that we don't change the existing behavior?